atomic_reactor/plugins/post_tag_and_push.py

```diff
@@ -20,24 +22,40 @@ class TagAndPushPlugin(PostBuildPlugin):
     key = "tag_and_push"
     is_allowed_to_fail = False
 
-    def __init__(self, tasker, workflow, **kwargs):
+    def __init__(self, tasker, workflow, registries):
         """
         constructor
 
         :param tasker: DockerTasker instance
         :param workflow: DockerBuildWorkflow instance
+        :param registries: dict, keys are docker registries, values are dicts containing per-registry
+                           parameters. Currently only the "insecure" optional boolean parameter
+                           is supported which controls whether pushes are allowed over plain HTTP.
         """
         # call parent constructor
         super(TagAndPushPlugin, self).__init__(tasker, workflow)
-        self.log.warning("ignoring arguments '%s'", kwargs)
+
+        self.registries = deepcopy(registries)
 
     def run(self):
         pushed_images = []
-        for registry in self.workflow.push_conf.docker_registries:
+
+        if not self.workflow.tag_conf.unique_images:
+            self.workflow.tag_conf.add_unique_image(self.workflow.image)
```

Comment:
So we don't add `self.workflow.image` if there are any unique images configured? (Just checking)